app/controllers/api/v1/Web.java

```diff
@@ -0,0 +1,1274 @@
+      taskObject.addProperty(JsonKeys.SEVERITY, task.severity.getText());
+      taskObject.add(JsonKeys.HEURISTICS_SUMMARY, heuristicsArray);
+      taskSummaryArray.add(taskObject);
+
+      jobResourceUsed += task.resourceUsed;
+      jobResourceWasted += task.resourceWasted;
+
+      if (jobSeverity.getValue() < task.severity.getValue()) {
+        jobSeverity = task.severity;
+      }
+      if (jobStartTime > task.startTime) {
+        jobStartTime = task.startTime;
+      }
+
+      if (jobEndTime < task.finishTime) {
+        jobEndTime = task.finishTime;
+      }
+    }
+
+    JsonArray taskSeverity = new JsonArray();
+
+    for (Map.Entry<Severity, Long> entry : taskSeverityCount.entrySet()) {
+      JsonObject severityObject = new JsonObject();
+      severityObject.addProperty(JsonKeys.SEVERITY, entry.getKey().getText());
+      severityObject.addProperty(JsonKeys.COUNT, entry.getValue());
+      taskSeverity.add(severityObject);
+    }
+
+    jobRuntime = Utils.getTotalRuntime(results);
+    jobDelay = Utils.getTotalWaittime(results);
+    JsonObject data = new JsonObject();
+    data.addProperty(JsonKeys.ID, jobDefID);
+    data.addProperty(JsonKeys.USERNAME, username);
+    data.addProperty(JsonKeys.JOB_NAME, jobname);
+    data.addProperty(JsonKeys.JOB_TYPE, jobtype);
+    data.addProperty(JsonKeys.START_TIME, jobStartTime);
+    data.addProperty(JsonKeys.FINISH_TIME, jobEndTime);
+    data.addProperty(JsonKeys.RUNTIME, jobRuntime);
+    data.addProperty(JsonKeys.WAITTIME, jobDelay);
+    data.addProperty(JsonKeys.RESOURCE_USED, jobResourceUsed);
+    data.addProperty(JsonKeys.RESOURCE_WASTED, jobResourceWasted);
+    data.addProperty(JsonKeys.SEVERITY, jobSeverity.getText());
+    data.addProperty(JsonKeys.JOB_EXEC_ID, jobExecutionId);
+    data.addProperty(JsonKeys.JOB_DEF_ID, jobDefinitionId);
+    data.addProperty(JsonKeys.FLOW_EXEC_ID, flowExecutionId);
+    data.addProperty(JsonKeys.FLOW_DEF_ID, flowDefinitionId);
+    data.add(JsonKeys.TASKS_SUMMARIES, taskSummaryArray);
+    data.add(JsonKeys.TASKS_SEVERITY, taskSeverity);
+
+    JsonObject parent = new JsonObject();
+    parent.add(JsonKeys.JOBS, data);
+    return ok(new Gson().toJson(parent));
+  }
+
+  /**
+   * @param applicationId
+   * @return
+   * <pre>
+   *  {
+   *  "applications": {
+   *    "id": "application_id",
+   *    "username": "username",
+   *    "jobtype": "Pig",
+   *    "mapreducejobname": "mapreducejobname",
+   *    "starttime": 1471910835628,
+   *    "finishtime": 1471911099238,
+   *    "runtime": 263610,
+   *    "waittime": 46234,
+   *    "resourceused": 101382144,
+   *    "resourcewasted": 15993417,
+   *    "severity": "Moderate",
+   *    "jobtracker": "jobtracker_address",
+   *    "jobexecid": "jobexecutionid",
+   *    "jobdefid": "jobdefinitionid",
+   *    "flowexeid": "flowexecutionid",
+   *    "flowdefid": "flowdefinitionid",
+   *    "yarnappheuristicresults": [
+   *      {
+   *        "name": "Mapper Data Skew",
```

Comment:
I think it's all right to be descriptive here. The developer editing the rest end point should know exactly what he's changing.